service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/AllocationDecisionControllerAboutToStartTest.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(AllocationDecisionController.class)
+@OverrideAutoConfiguration(enabled = true)
+class AllocationDecisionControllerAboutToStartTest {
+
+    private final AllocationDecisionController controller = new AllocationDecisionController();
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    void shouldAddYesToMissingAllocationDecision() throws Exception {
+
+        CallbackRequest request = CallbackRequest.builder().caseDetails(CaseDetails.builder()
+            .data(ImmutableMap.<String, Object>builder()
+                .put("allocationProposal","allocation proposal present")
+                .build()).build())
+            .build();
+
+        MvcResult response = mockMvc
+            .perform(post("/callback/allocation-decision/about-to-start")
+                .header("authorization", AUTH_TOKEN)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(MAPPER.writeValueAsString(request)))
+            .andExpect(status().isOk())
+            .andReturn();
```

Comment:
Can you extract that Mock invocation to the separate method? It's a bit too much code repeated and makes it harder to read. Maybe it can return callbackResponse for smaller tests.

Reply:
done :)